goldstone/core/tests.py

```diff
@@ -332,6 +335,83 @@ def test_alert_search_save(self):
         self.assertEqual(esmock.call_count, 1)
 
 
+class AlertSearchSQLQueryModelTests(SimpleTestCase):
+
+    """
+        Write a test to mock insertion of vulnerability data into the table
+        and test search(), search_recent() and return_query_results()
+        functionality
+    """
+    sample_vulnerability = [{"ossa_id": "OSSA-2014-020",
+                            "link": "http://abc.com/ossa/2014020",
+                             "processed": True,
+                             "name": "OpenStack upgrade causes Memory leak",
+                             "description": "It's a real mess",
+                             "affects": "Swift: 1.11.0 to 1.13.1",
+                             "patches": "I'm depending on you, son",
+                             "references": "Impeccable",
+                             "notes": "A",
+                             "cve": "CVE-2014-3497",
+                             "notice_date": "2014-06-19",
+                             "created": "2015-01-30 00:00:12.017634+00",
+                             "updated": "2016-01-30 00:00:12.017634+00"}]
+
+    @mock.patch('goldstone.core.models.AlertSearchSQLQuery')
+    @mock.patch('django.db.connection.cursor')
+    @mock.patch('django.db.backends.postgresql_psycopg2')
+    def test_check_sql_search_call(self, asql_mock, cur_mock, mock_db):
+
+        sample_db_entry = {"ossa_id": "OSSA-2014-020",
+                           "link": "http://abc.com/ossa/2014020",
+                           "processed": True,
+                           "name": "OpenStack upgrade causes Memory leak",
+                           "description": "It's a real mess",
+                           "affects": "Swift: 1.11.0 to 1.13.1",
+                           "patches": "Yet to be released",
+                           "references": "Impeccable",
+                           "notes": "A",
+                           "cve": "CVE-2014-3497",
+                           "notice_date": "2014-06-19"}
+
+        start_time = '2016-01-30 00:00:12.017634+00'
+        end_time = '2016-02-01 00:00:12.017634+00'
+
+        sql_db_entry = mock.MagicMock(spec=sample_db_entry)
+        sql_db_entry.name = 'name'
+        sql_db_entry._state = mock.MagicMock()
+        sql_db_entry.savedsearch_ptr_id = mock.MagicMock()
+
+        asql_mock.objects = mock.MagicMock()
+        asql_mock.objects.all = mock.MagicMock()
+        asql_mock.objects.all.return_value = [sql_db_entry]
+
+        cur_mock = mock.MagicMock(spec=connection.cursor())
+        cur_ec_mock = mock.MagicMock(spec=cur_mock)
+        cur_ec_mock.execute = mock.MagicMock(return_value=[sample_db_entry])
+
+        # simulate results of a search() call with 1 db entry
+        as_sql_inst = AlertSearchSQLQuery()
```

Comment:
The thing I do want to test is just the two functions  search & search_recent calls of AlertSearchSQLQuery (which I have not mocked). 

I'll verify this again.